### DISCUSSION TITLE
Can someone ELI5 how you handle updates and machine changes?

### DISCUSSION THREAD

**Discussion by moose-on-road** [grblHAL/core#574] [Q&A]
Last December I changed out my Uno/grbl 1.1 for a pico/grblHAL. I remember it being a slog to troubleshoot inputs and network settings but I got it working. Today I just installed a new spindle and I wanted to find my previous files so I could browse though the code I hadn't looked at or thought about in 8 months. This lead to me trying to view differences between my files and the master on github desktop, there were a number of unavoidable pop ups about uncommitted changes and detached heads and upstream something or other. With everything seeming to indicate that I needed to save my local files and commit them to Github, so that is what I tried to do. Instead github desktop overwrote my local files with the latest generic RP2040 driver, and somehow this is irreversable. Now I have no record of the modifications I made last december and have no idea how long it is going to take me to get the CNC running with the new spindle. Should I just put the old spindle back on and then never try to modify or update the controller ever again? 

How am I SUPPOSED to merge updated grblHAL code and my modified code in the future?

**Comment by terjeio** [comment]
> How am I SUPPOSED to merge updated grblHAL code and my modified code in the future?

I am no git expert, and I have a hard time understanding how it works coming from Subversion where an update automatically merges without overwriting my own changes. Perhaps [this search](https://duckduckgo.com/?t=ffab&q=git+merge+incoming&ia=web) contains useful leads?
